app/src/main/java/com/example/tbuckley/brakes/MainActivity.java

```diff
@@ -69,27 +109,41 @@ protected void onNewIntent(Intent intent) {
     public void handleIntent(Intent intent) {
         String action = intent.getAction();
 
-        if(Intent.ACTION_VIEW.equals(action)) {
+        if (Intent.ACTION_VIEW.equals(action)) {
             mWebView.setVisibility(View.VISIBLE);
             mTextView.setVisibility(View.GONE);
+            mUrlNavigationToggle.setVisibility(View.GONE);
             Uri data = intent.getData();
 
+            // Allow loading a URL from a VIEW Intent.
+            mPreventUrlLoading = false;
             mClearHistory = true;
             Log.d(TAG, "loadUrl: " + data.toString());
             mWebView.loadUrl(data.toString());
             mWebView.clearHistory();
         } else {
             mWebView.setVisibility(View.GONE);
             mTextView.setVisibility(View.VISIBLE);
+            mUrlNavigationToggle.setVisibility(View.VISIBLE);
         }
     }
 
     @Override
     public void onBackPressed() {
-        if(mWebView.canGoBack()) {
+        if (mWebView.canGoBack()) {
             mWebView.goBack();
         } else {
             super.onBackPressed();
         }
     }
+
+    @Override
+    public boolean onTouch(View view, MotionEvent motionEvent) {
```

Comment:
Should mPreventUrlLoading instead be set to the preference after the first page is loaded, eg. in onPageFinished?

Reply:
onPageStarted and shouldOverrideUrlLoading are too early because some web pages have redirects. onPageFinished is too late because some pages take a long time to load. The touch is the best one I have found so far.

Reply:
Ah, makes sense!